mllib/src/main/scala/org/apache/spark/mllib/regression/IsotonicRegression.scala

```diff
@@ -124,6 +130,71 @@ class IsotonicRegressionModel (
       predictions(foundIndex)
     }
   }
+
+  override def save(sc: SparkContext, path: String): Unit = {
+    val data = IsotonicRegressionModel.SaveLoadV1_0.Data(boundaries, predictions, isotonic)
+    IsotonicRegressionModel.SaveLoadV1_0.save(sc, path, data)
+  }
+
+  override protected def formatVersion: String = "1.0"
+}
+
+object IsotonicRegressionModel extends Loader[IsotonicRegressionModel] {
+
+  import org.apache.spark.mllib.util.Loader._
+
+  private  object SaveLoadV1_0 {
+
+    def thisFormatVersion: String = "1.0"
+
+    /** Hard-code class name string in case it changes in the future */
+    def thisClassName: String = "org.apache.spark.mllib.regression.IsotonicRegressionModel"
+
+    /** Model data for model import/export */
+    case class Data(boundaries: Array[Double], predictions: Array[Double], isotonic: Boolean)
```

Comment:
It would be easier to inspect the data file if we put each interval as a record. For example:

boundary | prediction
--------|--------------
0.0 | -1.0
1.0 | 0.5
2.0 | 1.0

We can save `isotonic` as a value in the metadata.